File: SDKs/amazonaws.com/pinpoint/2016-12-01/python/src/sdk/models/shared/simpleemail.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class SimpleEmail:
    r"""SimpleEmail
    Specifies the contents of an email message, composed of a subject, a text part, and an HTML part.
    """
    
    html_part: Optional[SimpleEmailPart] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('HtmlPart') }})
    subject: Optional[SimpleEmailPart] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Subject') }})
    text_part: Optional[SimpleEmailPart] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('TextPart') }})
    

File: SDKs/amazonaws.com/pinpoint/2016-12-01/typescript/src/sdk/models/shared/simpleemail.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { SimpleEmailPart } from "./simpleemailpart";



// SimpleEmail
/** 
 * Specifies the contents of an email message, composed of a subject, a text part, and an HTML part.
**/
export class SimpleEmail extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=HtmlPart" })
  htmlPart?: SimpleEmailPart;

  @SpeakeasyMetadata({ data: "json, name=Subject" })
  subject?: SimpleEmailPart;

  @SpeakeasyMetadata({ data: "json, name=TextPart" })
  textPart?: SimpleEmailPart;
}
